synaptic_reconstruction/inference/util.py

```diff
@@ -44,3 +127,88 @@ def get_prediction_torch_em(
     if verbose:
         print("Prediction time in", time.time() - t0, "s")
     return pred
+
+
+# TODO we also need to support .rec files ...
+def _get_file_paths(input_path, ext=".mrc"):
+    if not os.path.exists(input_path):
+        raise Exception(f"Input path not found {input_path}")
+
+    if os.path.isfile(input_path):
+        input_files = [input_path]
+        input_root = None
+    else:
+        input_files = sorted(glob(os.path.join(input_path, "**", f"*{ext}"), recursive=True))
+        input_root = None
+
+    return input_files, input_root
+
+
+def _load_input(img_path, extra_files, i):
+    # Load the mrc data
+    with open_file(img_path, "r") as f:
+        input_volume = f["data"][:]
+    assert input_volume.ndim == 3
+
+    # For now we assume this is always tif.
+    if extra_files is not None:
+        extra_input = imageio.imread(extra_files[i])
+        assert extra_input.shape == input_volume.shape
+        input_volume = np.stack(input_volume, extra_input)
+
+    return input_volume
+
+
+def inference_helper(
```

Comment:
I have refactored the prediction logic here, so that we don't need to duplicate this for each new inference file. This should also cover the use case of cristae inference using the `extra_input_path` logic.